doc/1.4/language.md

```diff
@@ -3815,6 +3828,271 @@ independent method callback(int i, void *aux) {
 }
 ```
 
+### Template-Qualified Method Implementation Calls
+Every object, as well as [every template type](#templates-as-types), has a
+`templates` member to allow for calling _particular_ implementations of that
+object's methods, as opposed to only the final overriding implementations
+that are reachable directly. Specifically, `templates` allows for invoking any
+particular implementation as provided by a specified template instantiated by
+the object. Such invocations are called _template-qualified method
+implementation calls_, and are made as follows:
+```
+template t {
+    method m() default {
+        log info: "implementation from 't'"
+    }
+}
+
+template u is t {
+    method m() default {
+        log info: "implementation from 'u'"
+    }
+}
+
+group g is u {
+    method m() {
+        log info: "final implementation"
+    }
+}
+
+method call_ms() {
+    // Logs "final implementation"
+    g.m();
+    // Logs "implementation from 'u'"
+    g.templates.u.m();
+    // Logs "implementation from 't'"
+    g.templates.t.m();
+}
+```
+
+Template-qualified method implementation calls are primarily meant as a way
+for an overriding method to reference overridden implementations, *even when*
+the implementations are provided by hierarchically unrelated templates such that
+`default` can't be used (see [Resolution of
+overrides](#resolution-of-overrides).) In particular, this typically allows for
+ergonomically resolving conflicts introduced when multiple orthogonal templates
+are instantiated, as long as all conflicting implementations are overridable,
+and one of the following is true:
+* The implementations can be combined together by calling each one of them, as
+  long as that can be done without risking e.g. side-effects being duplicated.
+* The implementations can be combined by choosing one particular template's
+  implementation to invoke (typically the one most complex), and then adding
+  code around that implementation call in order to replicate the behaviour of
+  the implementations of the other templates. Ideally, the other templates would
+  provide methods that may be leveraged so that their behaviour may be
+  replicated without the need for excessive boilerplate.
+
+The following is an example of the first case:
+```
+template simple_write_alter is write {
+    method write(uint64 written) {
+        default(alter_write(written));
+    }
+
+    method alter_write(uint64 curr, uint64 written) -> (uint64);
```

Comment:
inconsistency: alter_write vs simple_alter_write. I suggest using alter_write consistently, `simple` is redundant

Reply:
:thumbsup:

Reply:
Done.